source/exponential_backoff_retry_strategy.c

```diff
@@ -199,7 +203,9 @@ static uint64_t s_compute_deccorelated_jitter(struct exponential_backoff_retry_t
         return s_compute_full_jitter(token);
     }
 
-    return s_random_in_range(token->backoff_scale_factor_ns, aws_mul_u64_saturating(last_backoff_val, 3), token);
+    return aws_min_u64(
+        token->maximum_backoff_ns,
+        s_random_in_range(token->backoff_scale_factor_ns, aws_mul_u64_saturating(last_backoff_val, 3), token));
 }
```

Comment:
you'll want to do `min(X, max_backoff)` before you do `random_in_range()`

if you clamp it after the random(), then you'll have a random number 0->infinity, clamped to MAX. So at some point that random number will almost always be insanely large, so it will pretty much always get clamped to MAX, so they're won't be any randomness anymore

Reply:
I am not sure what the right answer is, but I copied the formula from https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/. 

Some thoughts:
- A cap is also in place on the maximum number of retries, which should mitigate instances where the random number becomes exceedingly large.
- Another thing is that `last_backoff_val` cannot be greater than `maximum_backoff`. So at most, it would be `3 * maximum_backoff` and there will be a 66% chance of getting the `maximum_backoff` again. There will still be an element of randomness.

Reply:
the blog post formula for decorrelated jitter is:
`sleep = min(cap, random_between(base, sleep * 3))`

they do random before the clamp, like I said

Reply:
agreed, it doesn't super duper matter, since we rolled a bunch of random sleeps to get here, so we're already "jittered" from whatever other clients are doing

but it's weird to have a random number generator which, after running for a while, starts returning the same number 2/3 of the time

???